cmd/datamon/cmd/config_create.go

```diff
@@ -2,56 +2,74 @@ package cmd
 
 import (
 	"io/ioutil"
+	"log"
 	"os"
-	"os/user"
 	"path/filepath"
 
 	"gopkg.in/yaml.v2"
 
 	"github.com/spf13/cobra"
 )
 
-const datamonDir = ".datamon2"
+const (
+	datamonDir = ".datamon2"
+	configFile = "datamon"
+	configExt  = ".yaml"
+)
 
 var configGen = &cobra.Command{
 	Use:   "create",
 	Short: "Create a config",
 	Long: `Create a config to use for datamon to hold flags that do not
 change.
 
-The configuration file will be placed in $HOME/` + datamonDir + `/datamon.yaml`,
+The configuration file will be placed in ` +
+		filepath.Join("$HOME", datamonDir, configFile+configExt),
 	Example: `# Replace path to gcloud credential file. Use absolute path
 % datamon config create --credential /Users/ritesh/.config/gcloud/application_default_credentials.json,
+config file created in /Users/ritesh/.datamon2/datamon.yaml
 
 # Replace path to gcloud credential file (use absolute path here)
-% datamon config create --credential /Users/ritesh/.config/gcloud/application_default_credentials.json`,
+% datamon config create --credential /Users/ritesh/.config/gcloud/application_default_credentials.json
+config file created in /Users/ritesh/.datamon2/datamon.yaml
+`,
 	Run: func(cmd *cobra.Command, args []string) {
 		_, err := paramsToContributor(datamonFlags)
 		if err != nil {
 			wrapFatalln("contributor datamonFlags present", err)
 			return
 		}
-		user, err := user.Current()
-		if user == nil || err != nil {
-			wrapFatalln("Could not get home directory for user", nil)
+		home, err := os.UserHomeDir()
+		if err != nil {
+			wrapFatalln("could not get home directory for user", err)
 			return
 		}
+
 		config := CLIConfig{
 			Config:     datamonFlags.core.Config,
 			Context:    datamonFlags.context.Descriptor.Name,
 			Credential: datamonFlags.root.credFile,
 		}
-		o, e := yaml.Marshal(config)
-		if e != nil {
-			wrapFatalln("serialize config to yaml", e)
+		o, err := yaml.Marshal(config)
```

Comment:
could there be a
```
switch configExt {
case ".yaml":
  o, err := <serialize-yaml>
default:
  wrapFatalln("implementation error", errors.New("unknown config format"))
```

here?  i mean, what's the point of keeping the extension as a constant if that constant remains separate and apart from the serialization?

Reply:
I think i ll bring serialization / deserialization together for improved readability